### PR TITLE
Use of the kubeless http trigger instead of a custom created ingress

### PR DESCRIPTION
**Pull request by faxioman**
This pull request enables the support for the Kubeless http trigger.
The code related to the creation/deletion of a custom ingress is removed.
All the options of the previous release of the serverless-kubeless plugin are preserved with this exceptions:

- the ingress.tlsConfig option is replaced by a ingress.tlsSecretName option, same as the --tls-secret flag when using the kubeless cli
- a new option ingress.enableTlsAcme is added, same as the --enableTLSAcme flag when using the kubeless cli
- a new option ingress.basicAuthSecretName is added, same as the --basic-auth-secret flag when using the kubeless cli
- a new boolean option ingress.cors is added, same as the --cors-enable flag when using the kubeless cli

This pull request includes a workaround for the issue https://github.com/kubeless/kubeless/issues/1124

### CODE REVIEW

**Comment by andresmgot** [comment]
Thanks @faxioman, I will review this shortly. It also seems that the CI is failing in the test that creates the ingress/http trigger:

```
    http-custom-path
      1) should return a "hello world" in a subpath
```

We probably need to update that, I will let you know once I have more info.

**Comment by faxioman** [comment]
Thanks @andresmgot. I was checking the failing integration test ... I could check it out. Does that work for you?

**Comment by andresmgot** [comment]
sure, that'd be great

**Comment by faxioman** [comment]
Ok, I also need to patch the get-info.js module.

**Comment by faxioman** [comment]
@andresmgot I made some fixes and now all the integration tests are passing.
I have a problem in a unit test though. The test is "should return the URL in case a path is specified". The URL is now available for an http trigger doing "serverless info"... so my problem is the test :)
I really can't figure out how to fix this test in order to pass it. Can you help me? ;)
Thanks!

File: test/kubelessInfo.test.js [hunk 1]
```diff
@@ -139,31 +139,6 @@ describe('KubelessInfo', () => {
         .get(`/apis/kubeless.io/v1beta1/namespaces/${f.namespace}/functions/${f.id}`)
         .reply(200, _.find(allFunctions, (ff) => ff.metadata.name === f.id));
     });
-
-
-    // Mock call to get.ingress
-    _.each(functions, f => {
```
Comment (andresmgot) [inline test/kubelessInfo.test.js:145]: I believe this the cause of the issue you are facing with the test. This is just a mock that returns the ingress if requested, why did you remove it?

Comment (faxioman) [reply test/kubelessInfo.test.js:145]: I removed it because the ingress is now created by kubeless ... so I think the serverless-kubeless plugin is not more responsible for the ingress creation. Right?

Comment (faxioman) [reply test/kubelessInfo.test.js:145]: I mean, the kubeless plugin is unaware of the ingress created by kubeless.

Comment (andresmgot) [reply test/kubelessInfo.test.js:145]: yeah, well, the result at the end of the day is the same. An Ingress with the name of the function is created so that's what this mock is doing.

If you want to remove this, you need to create the mock to return the HTTP trigger instead and change the `getInfo` code to request the HTTP trigger instead. I think that is more complicated to do so, that's why I suggest to just leave this as it was but if you want to run the extra mile and adapt that code, that's also fine ;)

Comment (faxioman) [reply test/kubelessInfo.test.js:145]: Ok ;)
We'll catch later.

Comment (faxioman) [reply test/kubelessInfo.test.js:145]: “Later” is the new “next week” 😏 Sorry For the delay @andresmgot. I will work on this pr next monday.

**Comment by faxioman** [comment]
@andresmgot now the getInfo method is based on the created http trigger.
I see an error in travis log ... but I think is unrelated to my changes.

File: lib/deploy.js [hunk 2]
```diff
@@ -434,6 +434,49 @@ function deployScheduleTrigger(name, namespace, schedule, options) {
...
+      labels: {
+        'created-by': 'kubeless',
+      },
+      annotations: Object.assign(baseAnnotations, options.ingress.additionalAnnotations),
+    },
+    spec: {
+      'basic-auth-secret': options.ingress.basicAuthSecretName || '',
+      'cors-enable': options.ingress.cors || false,
+      'function-name': name,
+      gateway: options.ingress.class || 'nginx',  // not working: https://github.com/kubeless/kubeless/issues/1124
```
Comment (andresmgot) [inline lib/deploy.js:461]: the comment is incorrect, right? it now works with these changes?

Comment (faxioman) [reply lib/deploy.js:461]: Unfortunately no. The gateway option is ignored by the kubeless controller, as reported in this issue: https://github.com/kubeless/kubeless/issues/1124
I added the gateway option so when the fix will be relased it will works as expected but until then, I added a workaround that adds the ingress class annotation.

File: lib/deploy.js [hunk 3]
```diff
@@ -434,6 +434,49 @@ function deployScheduleTrigger(name, namespace, schedule, options) {
     });
 }
 
+function deployHttpTrigger(name, namespace, path, options) {
+  if (options.ingress.enableTlsAcme && options.ingress.tlsSecretName) {
+    throw new Error('Cannot specify both enableTlsAcme and tlsSecretName');
+  }
+  // fix for issue https://github.com/kubeless/kubeless/issues/1124
```
Comment (faxioman) [inline lib/deploy.js:441]: This is the workaround for the issue https://github.com/kubeless/kubeless/issues/1124

**Comment by faxioman** [comment]
@andresmgot version bumped

File: lib/deploy.js (hunk 2 above)
Comment (andresmgot) [reply lib/deploy.js:461]: ah, right, that was an issue in the controller
